src/main/scala/com/fortysevendeg/macroid/extras/ViewTweaks.scala

```diff
@@ -0,0 +1,343 @@
+      popupMenu.setOnMenuItemClickListener(new OnMenuItemClickListener {
+        override def onMenuItemClick(item: MenuItem): Boolean = onMenuItemClickListener(item)
+      })
+      popupMenu.inflate(menu)
+      popupMenu.show()
+    }
+
+  def vPopupMenuShow(menu: Seq[String], onMenuItemClickListener: (MenuItem) => Boolean)(implicit contextWrapper: ContextWrapper) =
+    Tweak[W] { view =>
+      val popupMenu = new PopupMenu(contextWrapper.bestAvailable, view)
+      popupMenu.setOnMenuItemClickListener(new OnMenuItemClickListener {
+        override def onMenuItemClick(item: MenuItem): Boolean = onMenuItemClickListener(item)
+      })
+      menu.zipWithIndex foreach {
+        case (item, order) => popupMenu.getMenu.add(0, 0, order, item)
+      }
+      popupMenu.show()
+    }
+
+  def vListPopupWindowShow(
+    layout: Int,
+    menu: Seq[String],
+    onItemClickListener: (Int) => Unit,
+    width: Option[Int] = None,
+    height: Option[Int] = None)(implicit contextWrapper: ContextWrapper) =
+    Tweak[W] { view =>
+      val listPopupWindow = new ListPopupWindow(contextWrapper.bestAvailable)
+      listPopupWindow.setAdapter(new ArrayAdapter(contextWrapper.bestAvailable, layout, menu.toArray))
+      listPopupWindow.setAnchorView(view)
+      width foreach listPopupWindow.setWidth
+      height foreach listPopupWindow.setHeight
+      listPopupWindow.setModal(true)
+      listPopupWindow.setOnItemClickListener(new OnItemClickListener {
+        override def onItemClick(parent: AdapterView[_], view: View, position: Int, id: Long): Unit = {
+          onItemClickListener(position)
+          listPopupWindow.dismiss()
+        }
+      })
+      listPopupWindow.show()
+    }
+
+  def vSnackbarShort(res: Int) = Tweak[W] { view =>
+    Ui(Snackbar.make(view, res, Snackbar.LENGTH_SHORT).show()).run
+  }
+
+  def vSnackbarLong(res: Int) = Tweak[W] { view =>
+    Ui(Snackbar.make(view, res, Snackbar.LENGTH_LONG).show()).run
+  }
+
+  def vSnackbarIndefinite(res: Int) = Tweak[W] { view =>
+    Ui(Snackbar.make(view, res, Snackbar.LENGTH_INDEFINITE).show()).run
+  }
+
+  def vSnackbarShort(message: String) = Tweak[W] { view =>
+    Ui(Snackbar.make(view, message, Snackbar.LENGTH_SHORT).show()).run
+  }
+
+  def vSnackbarLong(message: String) = Tweak[W] { view =>
+    Ui(Snackbar.make(view, message, Snackbar.LENGTH_LONG).show()).run
+  }
+
+  def vSnackbarIndefinite(message: String) = Tweak[W] { view =>
+    Ui(Snackbar.make(view, message, Snackbar.LENGTH_INDEFINITE).show()).run
+  }
+
+  def vSnackbarIndefiniteAction(res: Int, buttonText: Int, f: () => Unit) = Tweak[W] { view =>
+    Ui(Snackbar.make(view, res, Snackbar.LENGTH_INDEFINITE).setAction(buttonText, new OnClickListener {
+      override def onClick(v: View): Unit = f()
+    }).show()).run
+  }
+
+
+}
+
+object ViewCompatTweaks {
+  type W = View
+
+  def vcElevation(elevation: Float): Tweak[W] = Tweak[W] (ViewCompat.setElevation(_, elevation))
+
```

Comment:
We should have the `view` variable in the first line and use the arrow symbol instead of `=>`